src/Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs

```diff
@@ -170,55 +205,141 @@ public IList<NameValueHeaderValue> Parameters
         /// </summary>
-        public bool MatchesAllTypes
+        /// <example>
+        /// For the media type <c>"application/vnd.example+json"</c>, the property gives the value
+        /// <c>"json"</c>.
+        /// </example>
+        public StringSegment SubTypeSuffix
         {
             get
             {
-                return MediaType.Equals("*/*", StringComparison.Ordinal);
+                var subTypeSuffixLength = _mediaType.LastIndexOf(PlusCharacter);
+                if (subTypeSuffixLength == -1)
+                {
+                    return default(StringSegment);
+                }
+                return _mediaType.Subsegment(subTypeSuffixLength + 1);
             }
         }
 
+
         /// <summary>
-        /// SubType = "*"
+        /// Get a <see cref="IList{T}"/> of facets of the <see cref="MediaTypeHeaderValue"/>. Facets are a
+        /// period separated list of StringSegments in the <see cref="SubTypeWithoutSuffix"/>.
+        /// See <see href="https://tools.ietf.org/html/rfc6838#section-3">The RFC documentation on facets.</see>
         /// </summary>
-        public bool MatchesAllSubTypes
+        /// <example>
+        /// For the media type <c>"application/vnd.example+json"</c>, the property gives the value:
+        /// <c>{"vnd", "example"}</c>
+        /// </example>
+        public IList<StringSegment> Facets
         {
             get
             {
-                return SubType.Equals("*", StringComparison.Ordinal);
+                return SubTypeWithoutSuffix.Split(new char[] { PeriodCharacter } ).ToList();
             }
         }
 
+        /// <summary>
+        /// Gets whether this <see cref="MediaTypeHeaderValue"/> matches all types.
+        /// </summary>
+        public bool MatchesAllTypes => MediaType.Equals(MatchesAllString, StringComparison.Ordinal);
+
+        /// <summary>
+        /// Gets whether this <see cref="MediaTypeHeaderValue"/> matches all subtypes.
+        /// </summary>
+        /// <example>
+        /// For the media type <c>"application/*"</c>, this property is <c>true</c>.
+        /// </example>
+        /// <example>
+        /// For the media type <c>"application/json"</c>, this property is <c>false</c>.
+        /// </example>
+        public bool MatchesAllSubTypes => SubType.Equals(WildcardString, StringComparison.Ordinal);
+
+        /// <summary>
+        /// Gets whether this <see cref="MediaTypeHeaderValue"/> matches all subtypes, ignoring any structured syntax suffix.
+        /// </summary>
+        /// <example>
+        /// For the media type <c>"application/*+json"</c>, this property is <c>true</c>.
+        /// </example>
+        /// <example>
+        /// For the media type <c>"application/vnd.example+json"</c>, this property is <c>false</c>.
+        /// </example>
+        public bool MatchesAllSubTypesWithoutSuffix =>
```

Comment:
I don't mind having it in abstractions if it's generally useful, but maybe some of it should move to a ContentNegotiation class to separate the header from the processing.